include/seqan3/search/algorithm/detail/search.hpp

```diff
@@ -181,36 +181,32 @@ inline auto search_single(index_t const & index, query_t & query, configuration_
 template <typename index_t, typename queries_t, typename configuration_t>
 inline auto search_all(index_t const & index, queries_t && queries, configuration_t const & cfg)
 {
-    using cfg_t = remove_cvref_t<configuration_t>;
-    // return type: for each query: a vector of text_positions (or cursors)
-    // delegate params: text_position (or cursor). we will withhold all hits of one query anyway to filter
-    //                  duplicates. more efficient to call delegate once with one vector instead of calling
-    //                  delegate for each hit separately at once.
-    using text_pos_t = std::conditional_t<index_t::text_layout_mode == text_layout::collection,
-                                          std::pair<typename index_t::size_type, typename index_t::size_type>,
-                                          typename index_t::size_type>;
-    using hit_t = std::conditional_t<cfg_t::template exists<search_cfg::output<detail::search_output_index_cursor>>(),
-                                     typename index_t::cursor_type,
-                                     text_pos_t>;
-
-    if constexpr (std::ranges::forward_range<queries_t> && std::ranges::random_access_range<value_type_t<queries_t>>)
+    if constexpr (!std::ranges::forward_range<std::ranges::range_value_t<queries_t>>)
```

Comment:
What about a collection of input ranges? wouldn't this cause an infinite loop here?

Reply:
This interface is called internally and is a detail interface. The public interface that calls this does all necessary checks. Input ranges are not allowed in general. The only question is if this is a collection of collection or not. And if so it calls itself again by making it a collection of collection.

Reply:
Maybe add an assert anyway?

Reply:
The point is in my opinion the entire design should be and probably needs to be refactored for the parallel implementation of the search interface. Then we can put this altogether in a configuration class similar to the alignment interface instead of having functions calling functions calling functions where we need to add all redundant checks again and again. So for now, I would simply leave it as is.